### DISCUSSION TITLE
Global File Manager

### DISCUSSION THREAD

**Discussion by NotExpectedYet** [OctoFarm/OctoFarm#219] [Feature Request]
On server file storage for OctoFarm. 

Allowing organisation, gcode scanning and parsing, samba share for easy file upload, ui upload, printer designation (from file name or selected)

**Comment by eovnu87435ds** [comment]
I definitely like the sound of this. With even just 2 identical printers, it's annoying to have to upload the same gcode twice. Multi-upload helps, but without the ability to multi-upload to subfolders, it could still be difficult to organize files. 

It would be really cool if octofarm mimicked the octoprint API for uploads, so that slicers that have a native "upload to octoprint" option could upload to octofarm instead.

**Reply by NotExpectedYet** [reply]
I can't remember why I took out the multi-upload to folders. I'm doing some work in a bit for 1.1.9 or the rest of... So I'll take a look at that see if I can't wangle it in. 

Does your slicer support repitier? It seems some do and that allows for printer selection so I was thinking to mimick that originally. That would be more for people with the same printer for each they own, myself personally would rather have the single upload. I'll have to have a look what they already offer currently as I'd rather not have to PR to the slicer to add it although native support would be cool! 

Regarding the current method, if you still upload and print to octoprint it should still grab the job information fine and continue tracking on OctoFarm as the websocket should monitor that.

**Reply by NotExpectedYet** [reply]
> I definitely like the sound of this. With even just 2 identical printers, it's annoying to have to upload the same gcode twice. Multi-upload helps, but without the ability to multi-upload to subfolders, it could still be difficult to organize files.
> 
> It would be really cool if octofarm mimicked the octoprint API for uploads, so that slicers that have a native "upload to octoprint" option could upload to octofarm instead.

Multi upload to folders is available and there's also a new Multi-Print option on views with a little more power.

**Reply by wicada** [reply]
Hello, I just wanted to know if something like this is in the works still? 

A central place to upload all the files that I could search for a file, select print, and then choose what printer to start it on (that's not currently printing) would be amazing! I would assume on the backend it would check to see if that file already exist on the octopi client and either start it if it does or upload it to the client and then start it if it does not. 

I currently use octofarm for the metrics, but this feature is preventing me from fulling using octofarm. I have a larger 100 printer farm and would be happy to sponsor for such a feature!
